Первый коммит

Stalnyetapki opened the pull request.

Stalnyetapki committed:
Первый коммит

Stalnyetapki committed:
Added first homework

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,159 @@
+    private static int catsNumber;
+    private String nameFood;
+    private String meow;
+    private static int i = 1;
```
Не называй так переменные, и я не совсем понял для чего она вообще.

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,159 @@
+    private String name;
+    private int weight;
+    private int satiety;
+    private static int catsNumber;
```
Обычно, визуально отделяют статику от обычных переменных, отступом. И название не совсем подходит, скорее catsCount наверное

Stalnyetapki replied:
По названию, если посмотреть в https://context.reverso.net/ и ввести "количество кошек", то предлагает варианты, как numbers of cats или amont of cats - вроде бы подходит в целом, но если это прям принципиально, то поменяю.
Отступ сделал.

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,159 @@
+    private int weight;
+    private int satiety;
+    private static int catsNumber;
+    private String nameFood;
```
foodName, если и писать) И тоже не понимаю, почему один из параметров кота, название еды...

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,159 @@
+    private String meow;
+    private static int i = 1;
+
+    public Kotik() {
```
Суть конструктора без параметров, чтобы там создавался пустой объект. И в задании описано, что потом, ты с помощью сеттеров задаешь его параметры.

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,159 @@
+        return catsNumber;
+    }
+
+    public static void setCatsNumber(int catsNumber) {
```
Не нужен сеттер у этого параметра, он не должен задаваться из вне.

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,159 @@
+    }
+
+    public void play() {
+        if (satiety > 0) {
```
Эта конструкция повторяется много раз, может выделить ее в 1 метод приватный и то что изменяется передавать параметром? А вообще эту проверку можно делать 1 раз в цикле, а не в каждом методе.

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,159 @@
+
+    public void eat(int satietyPoint, String nameFood) {
+        this.satiety += satietyPoint;
+        this.nameFood = nameFood;
```
Не нужен этот параметр котику, он ничего не дает, название еды можно использовать для вывода сообщения в консоль.

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,159 @@
+    }
+
+    public void eat(int satietyPoint) {
+        this.satiety += satietyPoint;
```
Тут надо использовать метод снизу с хардкодом, как в методе без параметров

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,159 @@
+
+    }
+
+    public void eat() {
```
Обычно, метод с полной реализацией помещают ниже всех, а над ним уже те, в которых он вызывается

Kuznetsov-DI commented on homework-1/src/Application.java:
```diff
@@ -0,0 +1,16 @@
+
+    public static void main(String[] args) {
+        Kotik catFirst = new Kotik("Vasya", 5, 10, "salam");
+        Kotik catSecond = new Kotik();
```
Параметры у второго котика должны задаваться сеттерами а не рандомом, так прописано в задаче.

Kuznetsov-DI commented on homework-1/homework-1.iml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Этот файл должен быть в гитигноре

Kuznetsov-DI replied:
После добавления в гитигнор, его надо удалить из гита

Stalnyetapki committed:
Fixed name of parameter in eat() method
Deleted unnecessary parameters from the Kotik class
Optimized methods play(), sleep(), chaseMouse(),doTygydyk(),sharpenClaws by moving repetitive code to the liveAnotherDay() method
Fixed constructor without parameters
Fixed class Application
Added homework-1.iml to .gitignore

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,124 @@
+                        break;
+                }
+            } else {
+                System.out.println("The cat " + name + " doesn't have enough energy");
```
Ты просто выводишь сообщение, но не кормишь кота. Глянь задание внимательнее

Stalnyetapki committed:
Delete homework-1/homework-1.iml

Stalnyetapki committed:
Changed the output message in eat() method. Now, it shows count of restored satiety points.
Fixed liveAnotherDay() method. Now, if the cat doesn't have enough energy to do any actions, the eat() method is executed, which restores some energy. Corrected the output message.

Stalnyetapki committed:
Merge remote-tracking branch 'origin/homework_1' into homework_1
